p2p/host/basic/basic_host.go

```diff
@@ -997,10 +998,83 @@ func (h *BasicHost) AllAddrs() []ma.Multiaddr {
 		}
 		finalAddrs = append(finalAddrs, observedAddrs...)
 	}
+	finalAddrs = inferWebtransportAddrsFromQuic(finalAddrs)
 
 	return dedupAddrs(finalAddrs)
 }
 
+var wtComponent = ma.StringCast("/webtransport")
+
+// inferWebtransportAddrsFromQuic infers more webtransport addresses from QUIC addresses.
+// This is useful when we discover our public QUIC address, but haven't discovered our public WebTransport addrs.
+// If we see that we are listening on the same port for QUIC and WebTransport,
+// we can be pretty sure that the WebTransport addr will be reachable if the
+// QUIC one is.
+func inferWebtransportAddrsFromQuic(in []ma.Multiaddr) []ma.Multiaddr {
+	// We need to check if we are listening on the same ip+port for QUIC and WebTransport.
+	// If not, there's nothing to do since we can't infer anything.
+
+	// Count the number of QUIC addrs, this will let us allocate just once at the beginning.
```

Comment:
Not sure if the plan to reduce allocs actually works out, `SplitLast` also allocates pretty heavily. No need to change it though, as this is temporary anyway.

Reply:
Does it? I thought it just pointed to slices?